### DISCUSSION TITLE
How to load local MIDI file dataset?

### DISCUSSION THREAD

**Discussion by cifkao** [salu133445/muspy#58] [Q&A]
Is there currently an easy way to load a local MIDI dataset from a directory? This seems like one of the most basic use cases for this dataset machinery. For example, I would like to use the Lakh MIDI dataset which I have, without downloading it again. Should a `MIDIFolderDataset` be added? Or should support for MIDI be added to `MusicDataset`?

**Comment by vandrw** [comment]
I have tried basing a class on `muspy.FolderDataset`, which seems to work fine for me.
```Python
class MidiFolderDataset(muspy.FolderDataset):
    _extension = "midi" # Change this based on your extension.

    def read(self, filename: Union[str, Path]) -> Music:
        """Read a file into a Music object."""
        return read_midi(self.root / filename)
```

You can now import the dataset:
```Python
music_dataset = MidiFolderDataset(data_dir, convert=True) #You can also avoid the convert parameter to enable on_the_fly conversion.
```
